src/common.c

```diff
@@ -69,7 +69,7 @@ int fi_poll_fd(int fd, int timeout)
 	fds.fd = fd;
 	fds.events = POLLIN;
 	ret = poll(&fds, 1, timeout);
-	return ret == -1 ? -errno : ret;
+	return ret == SOCKET_ERROR ? -errno : ret;
```

Comment:
Thanks for the patch!
I have a question regarding `errno`: I've found this link here that explains how to deal with error codes on Windows: https://msdn.microsoft.com/en-us/library/windows/desktop/ms737828(v=vs.85).aspx

For compatibility with BSD application, they suggest to include the following lines of code:
```
#include <windows.h>
#define errno WSAGetLastError()
```

I'm not sure if `errno` gets properly updated in case of failure. 
Shall we include this into `./include/windows/osd.h`?

Reply:
Oh, Thanks! Really very good point. It should be fixed for all places in the OFI code that can be compiled by Windows.
But, seems there are some drawbacks with this solution (for example, it produces on my test app the following warnings:
```
1>c:\users\dgladkov\documents\visual studio 2015\projects\consoleapplication1\consoleapplication1\source.cpp(14): warning C4005: 'errno': macro redefinition
1>  c:\program files (x86)\windows kits\10\include\10.0.10240.0\ucrt\errno.h(18): note: see previous definition of 'errno'
```

@hoopoepg added special function `ofi_sockerr()` in the #2066 PR. I think we should fix all places by utilization of `ofi_sockerr()` instead of `errno` .

@sydidelot Thanks again for your comment!

Reply:
right, ofi_sockerr was added to abstract error processing, unfortunately it could not be applied everywhere due to mix of error handling (system/ofi), but it is good idea to use it at least for system errors

thank you

Reply:
@hoopoepg Thank you!